prefect_email/credentials.py

```diff
@@ -90,11 +92,11 @@ class EmailServerCredentials(Block):
     _block_type_name = "Email Server Credentials"
     _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/3PcxFuO9XUqs7wU9MiUBMg/ca740e27815d15528373aced667f58b9/email__1_.png?h=250"  # noqa
 
-    username: str
-    password: SecretStr
-    smtp_server: Optional[Union[str, SMTPServer]] = SMTPServer.GMAIL
-    smtp_type: Optional[Union[str, SMTPType]] = SMTPType.SSL
-    smtp_port: Optional[int] = None
+    username: str = None
+    password: SecretStr = SecretStr("")
+    smtp_server: Union[str, SMTPServer] = SMTPServer.GMAIL
+    smtp_type: Union[str, SMTPType] = SMTPType.SSL
+    smtp_port: int = None
```

Comment:
```suggestion
    smtp_port: Optional[int] = None
```

Reply:
yeah... reading the typing doc page indicates Optional[x] == Union[x, None] so specifying the default as None as well is redundant. But there is a OTOH...  geez.

Reply:
As for the unit test... let's think this through. Testing the lack of supplied parameters is in effect a test that the Python interpreter does what it is supposed to - all we can do is validate that the attributes are set to the defaults which is a bit silly. If the defaults change, then the test needs changed which someone might overlook.. a minor hassle but why risk it? And there is a test that tests those two can be set correctly.  And what about tests that validate the others don't have to be set because there are defaults? That is a lot of pointless tests to run.

The other theoretically required test is to only validate just the last two simple lines of get_server() which means to properly evaluate that, we have to gin up SMTPMock to store whether or not login() has been called, then a routine to return that to a caller... then logically every other test SMTPMock is involved in should be double checked or modified to utilize this new capability of storing what SMTPMock routines have been called....  

That's a lot of boilerplate for something we can easily reason about - more code means possibly more maintenance work in the future - and the PM in me doesn't like that idea. If the logic in get_server() was complex, I'd be more inclined however - or if there was more complexity to this module than there is, given how username is only used on 3 lines of code in the entire module

And now I think I've spotted a possible something else related to username.  Sorry... :) I will have to refresh my memory of email specs first.

Reply:
In regards to: `If the defaults change, then the test needs changed..` I think that's the behavior we want. That would indicate there's a breaking change and we must not overlook that; without the test, it might be overlooked if we decide to switch back to required args.

Let me know if you'd like me to write those tests.

Reply:
Ok I added tests here: https://github.com/drfraser/prefect-email/pull/1. Let me know if that works for you!

Reply:
done. I think this branch is ready to merge